### ISSUE TITLE
onTabVisibility Callback Not Firing?

### ISSUE THREAD

**Issue by asentner** [uploadcare/uploadcare-widget#478]
Has anyone successfully got this callback to fire?

**Comment by optlsnd** [comment]
@asentner unfortunately, this callback is not working properly, as it has a bug which we're going to fix with the next widget update.

**Comment by optlsnd** [comment]
@asentner we've double-checked all the things, the onTabVisibility callback exists for internal purposes of the widget and it doesn't work in the way as expected :) We'll fix our docs and mention this. There's another callback that fires and provides the current tab's name when you switch the tab - dialog.progress().
```
widget.onDialogOpen(dialog => {
  dialog.progress(tab => {
    console.log(tab);
  });
});
```

**Comment by asentner** [comment]
@optlsnd Thanks for the update! That's what we ended up doing to get our code to fire. I noticed I can only get the `onTabVisibility` to run when using manual tab functions such as `showTab` and `hideTab`. Is this going to remain the case or will a future update fire this callback anytime the tab is opened? The reason I ask is `dialog.progress` is not very intuitive. May take someone a while to figure out that's the one to use.
